### ISSUE TITLE
Fill colors lost when reading/writing xlsx

### ISSUE THREAD

**Issue by ndench** [PHPOffice/PhpSpreadsheet#2075]
This is:

```
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected 

Reading an xlsx file that contains fill colors and writing it without making changes should result in cells having the same fill color.

### What is the current behavior?

Fill colors are being "lost". A closer look at the xlsx file `xl/styles.xml` hints that maybe the color is being changed to white.

Excerpt from original file (fill.xlsx):
```xml
    <fills count="3">
        <fill>
            <patternFill patternType="none"/>
        </fill>
        <fill>
            <patternFill patternType="gray125"/>
        </fill>
        <fill>
            <patternFill patternType="solid">
                <fgColor theme="7" tint="0.59996337778862885"/><bgColor indexed="64"/>
            </patternFill>
        </fill>
    </fills>
```

Excerpt from output file (output.xlsx):
```xml
    <fills count="3">
        <fill>
            <patternFill patternType="none"/>
        </fill>
        <fill>
            <patternFill patternType="gray125">
                <fgColor rgb="FFFFFFFF"/>
                <bgColor rgb="FF000000"/>
            </patternFill>
        </fill>
        <fill>
            <patternFill patternType="solid">
                <fgColor rgb="FFFFFFFF"/>
                <bgColor rgb="FFFFFFFF"/>
            </patternFill>
        </fill>
    </fills>
```

### What are the steps to reproduce?

```php
<?php

require __DIR__ . '/vendor/autoload.php';

// Create new Spreadsheet object
$spreadsheet     = PhpOffice\PhpSpreadsheet\IOFactory::load('fill.xlsx');

$writer = PhpOffice\PhpSpreadsheet\IOFactory::createWriter($spreadsheet, 'Xlsx');
$writer->save('output.xlsx');
```

Here is the original file, [fill.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/6445011/fill.xlsx), which I created by opening a blank spreadsheet, right clicking a cell -> format cells -> add a fill color.

Here is the [output.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/6445015/output.xlsx) which was created by running the above script.

### Which versions of PhpSpreadsheet and PHP are affected?

I'm running PHP 8.0.3 on Ubuntu 18.04.5 with "phpoffice/phpspreadsheet": "dev-master".
Then opening the resulting Excel file on Windows 10 with Excel 2016.

**Comment by MarkBaker** [comment]
Ug! Themes :-(

**Comment by ndench** [comment]
Ah right. Does PhpSpreadsheet not support themes? Is it possible to disable themes all together in Excel?

**Comment by MarkBaker** [comment]
Yes, but not well... the code for themes was implemented based on a lot of guesswork before MS published the OfficeOpenXML spec

**Comment by ndench** [comment]
Fair enough. Is there anything I can do to help? I'm not familiar with the spec but if you can point me in the general direction I'm happy to look into it.

**Comment by konovalovk** [comment]
I am using xlsx template with a cell having white color and some fill. Text and fill color are lost after writing to this cell and saving file.

```
	$workbook = PhpOffice\PhpSpreadsheet\IOFactory::load('template.xlsx');
	$workbook->getActiveSheet()->setCellValue('A12', 'white');
```

![image](https://user-images.githubusercontent.com/7712802/121024392-db26fd80-c7ac-11eb-8988-a0969403f562.png)

**Comment by MarkBaker** [comment]
@konovalovk Can you provide a copy of the file

**Comment by konovalovk** [comment]
[template.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/6609632/template.xlsx)
[ready.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/6609633/ready.xlsx)

yes, please

**Comment by TheFehr** [comment]
I do have a similar situation. For me the background fill stays correct but the text fill "resets" from white to black

**Comment by rcalpha** [comment]
I also have the same issue. I did the same test before reading this thread. Open and Save destoy some foreground and background color

**Comment by agaluf** [comment]
We have the same issue as well. Tests showed that some colors still remain (such as yellow background and black font), but many others were lost. We're not yet sure what that depends on.

However, tests also reveal that this bug was first introduced in 1.18, as previous releases seem to work without a problem on the same xlsx files. Both 1.18 and 1.19 are bugged. As a result, we temporarily rolled back to 1.17 and are looking for a fix.

More information may follow later today.

**Comment by thomas2411** [comment]
I think the problem still exists. I have 1.20.0 version and I have an Excel file where some of the cells are filled with #D9D9D9. When I open it with a `PhpOffice\PhpSpreadsheet\Reader\Xlsx` and then download it or save it as another file with `\PhpOffice\PhpSpreadsheet\Writer\Xlsx($spreadsheet)` fill color disappears.

**Comment by oleibman** [comment]
The PR that fixes this problem was merged after the release of 1.20. Can you test against master? If not, can you upload the file that has a problem?

**Comment by thomas2411** [comment]
I confirm @oleibman thank you. Any new when we can expect new relase with this fix?
